<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <!-- Number of threads used by OpenMP (for TEASER++)-->
  <env name="OMP_NUM_THREADS" value="12"/>

  <arg name="semantic_label_2_color_csv_filename"
    default="tesse_multiscene_office2_segmentation_mapping.csv"/>
  <arg name="gt_csv_filename"
    default="office_static_tfs_ros.csv"/>

  <arg name="target_object_label" default="Couch_1seat"/>
  <arg name="scene" default="office"/>

  <node name="object_db_node" pkg="object_db"
    type="server_node" output="screen">
    <param name="world_frame_id"                     value="map"/>

    <!-- Launch file for running on office2 scene with office1 seg-->
    <!-- refer to TEASER++ github for more information. See link below-->
    <!-- https://github.com/MIT-SPARK/TEASER-plusplus -->
    <!-- TEASER++ params -->
    <param name="barc2"                              value="1"/>
    <param name="noise_bound"                        value="0.1"/>
    <param name="estimate_scaling"                   value="false"/>
    <param name="rotation_algorithm"                 value="GNC_TLS"/>
    <param name="rotation_gnc_factor"                value="1.4"/>
    <param name="rotation_max_iterations"            value="100"/>
    <param name="rotation_cost_threshold"            value="0.000001"/>
    <param name="max_cique_find_exact"               value="true"/>

    <!-- Object matcher params -->
    <!-- Refer to PCL's documentation of Harris keypoint for more details -->
    <!-- https://pointclouds.org/documentation/classpcl_1_1_harris_keypoint3_d.html -->
    <!-- Number of threds used for the nearest-neighbor matcher -->
    <param name="matcher_num_threads"                 value="12"/>
    <!-- Radius for Harris 3D keypoint calculation -->
    <param name="matcher_radius"                      value="0.15"/>
    <!-- Non-maximal suppression threshold for Harris 3D keypoint detector -->
    <param name="matcher_nms_threshold"               value="0.0001"/>
    <!-- Use non-maximal suppression or not for Harris 3D keypoint -->
    <param name="matcher_use_nms"                     value="true"/>
    <!-- Use refine option or not for Harris 3D keypoint detector -->
    <param name="matcher_use_refine"                  value="true"/>

    <!-- Object Database Params -->
    <!-- Path to the object database -->
    <param name="object_db_path"      value="$(find object_db)/data/$(arg scene)_known_objects"/>
    <param name="target_object_label" value="$(arg target_object_label)"/>

    <!-- GT Data Params -->
    <!-- Path of the ground truth pose information csv file-->
    <param name="gt_csv_path"              value="$(find kimera_semantics_ros)/cfg/$(arg gt_csv_filename)"/>
    <!-- Path of the color to label mapping csv file-->
    <param name="semantic_label_csv_path"  value="$(find kimera_semantics_ros)/cfg/$(arg semantic_label_2_color_csv_filename)"/>

  </node>
</launch>
